runtime/tests/staking_reward.rs

```diff
@@ -269,116 +203,72 @@ fn staking_genesis_config_works() {
-						balance_amount + per_staking_reward
-					);
-				}
-			}
+			println!("{:?}", System::block_hash(0));
+			println!("{:?}", System::block_hash(1));
+			start_era(1);
 
-			// Total issuance for CPAY is inflated at the start of era 2, and that for CENNZ is unchanged.
-			start_session(7);
-			assert_eq!(Staking::current_era(), 2);
+			// Start with 0 transaction rewards
+			assert!(Rewards::transaction_fee_pot().is_zero());
 
-			let (total_payout, inflation_era_2) = Staking::current_total_payout(total_issuance + inflation_era_1);
-			assert_eq!(total_payout, 71_100);
-			assert_eq!(inflation_era_2, 189_600);
+			// Apply first extrinsic and check transaction rewards
+			println!("{:?}", System::block_hash(0));
+			println!("{:?}", System::block_hash(1));
+			frame_system::Module::<Runtime>::set_parent_hash([70u8; 32].into());
+			let r = Executive::apply_extrinsic(xt_1.clone());
+			println!("{:?}", r);
+			total_transfer_fee += extrinsic_fee_for(&xt_1);
+			assert_eq!(Rewards::transaction_fee_pot(), total_transfer_fee);
 
-			// validators should receive staking reward after new era
-			let per_staking_reward = total_payout / validator_len + per_staking_reward;
-			for (stash, _controller, _, _, _, _) in &validators {
-				assert_eq!(
-					<GenericAsset as MultiCurrency>::free_balance(&stash, Some(CENTRAPAY_ASSET_ID)),
-					balance_amount + per_staking_reward
-				);
-			}
+			// Apply second extrinsic and check transaction rewards
+			let r2 = Executive::apply_extrinsic(xt_2.clone());
+			println!("{:?}", r2);
+			total_transfer_fee += extrinsic_fee_for(&xt_2);
+			assert_eq!(Rewards::transaction_fee_pot(), total_transfer_fee);
 
-			let sessions_era_2 = vec![8, 9, 10, 11, 12];
-			for session in sessions_era_2 {
-				start_session(session);
-				assert_eq!(Staking::current_era(), 2);
-				// Total issuance for CENNZ is unchanged
-				assert_eq!(GenericAsset::total_issuance(CENNZ_ASSET_ID), total_issuance);
-				// Total issuance for CPAY remain the same within the same era
-				assert_eq!(
-					GenericAsset::total_issuance(CENTRAPAY_ASSET_ID),
-					total_issuance + inflation_era_1 + inflation_era_2
-				);
+			// Advancing sessions shouldn't change transaction rewards storage
+			advance_session();
+			assert_eq!(Rewards::transaction_fee_pot(), total_transfer_fee);
+			advance_session();
+			assert_eq!(Rewards::transaction_fee_pot(), total_transfer_fee);
 
-				// The balance of stash accounts remain the same within the same era
-				for (stash, _controller, _, _, _, _) in &validators {
-					assert_eq!(
-						<GenericAsset as MultiCurrency>::free_balance(&stash, Some(CENTRAPAY_ASSET_ID)),
-						balance_amount + per_staking_reward
-					);
-				}
-			}
+			// At the start of the next era (13th session), transaction rewards should be cleared (and paid out)
+			start_era(2);
+			advance_session();
+			assert_eq!(Staking::current_era(), 3);
+			assert!(Rewards::transaction_fee_pot().is_zero());
```

Comment:
Do we want to assert on where the fees are paid out to?

Reply:
This test only focuses on reward storage being changed and persisted across sessions and eras properly. There is another test for payout correctness